src/main/java/ch/njol/skript/expressions/ExprIndicesOfX.java

```diff
@@ -0,0 +1,184 @@
+	"set {_indices::*} to the last index of the value 100 in {_otherlist::*}",
+	"# {_indices::*} is now \"foo\"",
+	"",
+	"set {_positions::*} to the positions of the value 100 in {_otherlist::*}",
+	"# {_positions::*} is now 1, 3 and 5",
+	"set {_positions::*} to the positions of the value \"hi\" in {_otherlist::*}",
+	"# {_positions::*} is now 2 and 4",
+	"",
+	"set {_positions::*} to positions of \"mega\" in \"small\", \"mega\", \"mega\", \"medium\", \"small\" and \"mega\"",
+	"# {_positions::*} is now 2, 3 and 6"
+})
+@Since("INSERT VERSION")
+public class ExprIndicesOfX extends SimpleExpression<Object> {
+
+	static {
+		Skript.registerExpression(ExprIndicesOfX.class, Object.class, ExpressionType.COMBINED,
+			"[the] [1:first|2:last] (indices|index[es]) of [[the] value] %object% in %objects%",
+			"[the] [1:first|2:last] position[s] of [[the] value] %object% in %objects%"
+		);
+	}
+
+	private IndexType type;
+	private boolean position;
+	private Expression<?> value;
+	private Expression<?> objects;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (exprs[1].isSingle()) {
+			Skript.error("'" + exprs[1].toString(null, false) +
+					"' can only ever have one value at most, thus the 'indices of x in list' expression has no effect.");
+			return false;
+		}
+
+		if (!(exprs[1] instanceof Variable<?>) && matchedPattern == 0) {
+			Skript.error("'" + exprs[1].toString(null, false) +
+					"' is not a list variable. You can only get the indices of a list variable.");
+			return false;
+		}
+
+		position = matchedPattern == 1;
+		objects = LiteralUtils.defendExpression(exprs[1]);
+		type = IndexType.values()[parseResult.mark];
+		value = LiteralUtils.defendExpression(exprs[0]);
+
+		return LiteralUtils.canInitSafely(objects, value);
+	}
+
+	@Override
+	protected Object @Nullable [] get(Event event) {
+		Object value = this.value.getSingle(event);
+		if (value == null)
+			return new String[0];
+
+		List<Object> indices = new ArrayList<>();
+
+		int position = 1;
+		if (objects instanceof Variable<?> list) {
+			//noinspection unchecked
+			Map<String, Object> variable = (Map<String, Object>) list.getRaw(event);
+			if (variable == null)
+				return new String[0];
+
+			for (Map.Entry<String, Object> entry : variable.entrySet()) {
+				Object entryValue = entry.getValue();
+				// the value of {foo::1} when {foo::1::bar} is set is a map with a null key of the value {foo::1}
+				if (entryValue instanceof Map<?, ?> map)
+					entryValue = map.get(null);
+
+				if (entryValue.equals(value)) {
+					Object index = getPositionOrIndex(entry.getKey(), position, this.position);
+
+					if (type == IndexType.FIRST)
+						return new Object[]{index};
+
+					indices.add(index);
+				}
+				position++;
+			}
+		} else {
+			for (Object object : objects.getArray(event)) {
+				if (object.equals(value)) {
+					if (type == IndexType.FIRST)
+						return new Object[]{position};
```

Comment:
You're saying the return type is Integer or String, but you're returning Object[] here.
![Uploading image.png…]()